src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -187,8 +206,10 @@ static Permission updateResource(@NonNull Session session, @NonNull Permission p
                     logger.atWarn().kv("attributeName", attributeName)
                             .log("No attribute found for current session");
                 } else {
+                    // for ThingName support only, we can use .replaceAll()
+                    // to support additional policy variables in the future
                     resource = matcher.replaceFirst(policyVariableValue);
-
+                    matcher = POLICY_VARIABLE_PATTERN.matcher(resource);
```

Comment:
This works, but curious is there's a way to iterate over matches and replace without creating a new matcher every time?

Reply:
I think it's possible but it would require having a new variable for the updated resource bc of the way `matcher` works. `matcher` only has `replaceAll` (which would be fine for only using the ThingName variable) and `replaceFirst` (which is problematic because `matcher` will keep replacing the first instance of the policy variable in every iteration since it was instantiated with the original resource). With a new variable we would have to keep track of indices to append the value + parts of the resource that don't match, not sure if that's preferred (or if we should just `replaceAll` to only account for ThingName for now).